CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+# This is particularly useful to run ctests on libraries built on Windows
+# machines: tests, which are executables, are placed in the same folders of
+# dlls, which are treated as executables as well, so that they can properly
+# find the libraries to run. This is a because of missing RPATH on Windows.
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
+
+# To build shared libraries in Windows, we set CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS to TRUE.
+# See https://cmake.org/cmake/help/v3.4/variable/CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS.html
+# See https://blog.kitware.com/create-dlls-on-windows-without-declspec-using-new-cmake-export-all-feature/
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+# Under MSVC, we set CMAKE_DEBUG_POSTFIX to "d" to add a trailing "d" to library
+# built in debug mode. In this Windows user can compile, build and install the
+# library in both Release and Debug configuration avoiding naming clashes in the
+# installation directories.
+if(MSVC)
+    set(CMAKE_DEBUG_POSTFIX "d")
+endif()
+
+option(ENABLE_RPATH "Enable RPATH for this library" ON)
+mark_as_advanced(ENABLE_RPATH)
+include(AddInstallRPATHSupport)
+add_install_rpath_support(BIN_DIRS "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}"
+  LIB_DIRS "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}"
+  INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}"
+  DEPENDS ENABLE_RPATH
+  USE_LINK_PATH)
+
+#Function to automatize the process of creating a new library
+include(AddBiomechanicalAnalysisLibrary)
+
+add_subdirectory(src)
+
+#Creates the interface target BiomechanicalAnalysis::Framework that depends on all the targets.
+#By including the file "BiomechanicalAnalysis/Framework.h" it is possible to include all the headers automatically.
+include(AddBiomechanicalAnalysisFrameworkTarget)
+
+
+include(InstallBasicPackageFiles)
+get_property(FRAMEWORK_PRIVATE_DEPENDENCIES GLOBAL PROPERTY BiomechanicalAnalysisFramework_PrivateDependencies)
+get_property(FRAMEWORK_PUBLIC_DEPENDENCIES GLOBAL PROPERTY BiomechanicalAnalysisFramework_PublicDependencies)
+install_basic_package_files(${PROJECT_NAME}
+                            NAMESPACE BiomechanicalAnalysis::
+                            VERSION ${BiomechanicalAnalysisFramework_VERSION}
+                            COMPATIBILITY AnyNewerVersion
+                            TARGETS_PROPERTY BiomechanicalAnalysisFramework_TARGETS
+                            VARS_PREFIX ${PROJECT_NAME}
+                            NO_CHECK_REQUIRED_COMPONENTS_MACRO
+                            DEPENDENCIES ${FRAMEWORK_PUBLIC_DEPENDENCIES}
+                            PRIVATE_DEPENDENCIES ${FRAMEWORK_PRIVATE_DEPENDENCIES}
+                            OVERRIDE_MODULE_PATH ${CMAKE_INSTALL_PREFIX}/share/${PROJECT_NAME}/cmake)
```

Comment:
Just FYI, this is used in order to make sure that the targets that have ``BiomechanicalAnalysisFramework`` as a dependency, will also be able to find the dependencies of this library. This is necessary when a dependency needs a custom ``Find*.cmake``. For example, in ``blf`` we had this issue with ``cppad``. Hence, we installed the corresponding ``Findcppad.cmake`` file in the same location specified after ``OVERRIDE_MODULE_PATH``. In this way, the consumer target can find ``cppad`` as well because it can use the same ``Findcppad.cmake`` file. If you don't have any dependency that needs a custom ``Find*.cmake`` file, you can remove that line.

Reply:
Thanks @S-Dafarra for the explanation. Indeed, I have removed it for the time being.